django/db/backends/postgresql/creation.py

```diff
@@ -35,11 +35,16 @@ def _database_exists(self, cursor, database_name):
 
     def _execute_create_test_db(self, cursor, parameters, keepdb=False):
         try:
-            if keepdb and self._database_exists(cursor, parameters['dbname']):
+            exists = self._database_exists(cursor, parameters['dbname'])
+            if keepdb and exists:
                 # If the database should be kept and it already exists, don't
                 # try to create a new one.
                 return
-            super()._execute_create_test_db(cursor, parameters, keepdb)
+            if exists:
+                # In some cases, the user we connect is may not be permitted to create DBs.
+                cursor.execute('DROP OWNED BY CURRENT_USER;')
+            else:
```

Comment:
Is this `if..else` correct? 

Don't we want to 'DROP OWNED BY CURRENT_USER' and **then** call `super()._execute_create_test_db()`?

Reply:
The if/else is correct... from https://www.postgresql.org/docs/10/static/sql-drop-owned.html : 
```
DROP OWNED drops all the objects within the current database that are owned by one of the specified roles.
```

Mind you, this is assuming we're connected to the desired DB. We may be on the "postgres" db at this point. I was struggling with how to test this solidly, as well as on the last day of a conference :)

Reply:
Ah, OK. I understand. Thanks. (Good link!) 

> Mind you, this is assuming we're connected to the desired DB... 

Indeed. 🙂